src/library/ButtonGroup/ButtonGroup.js

```diff
@@ -0,0 +1,294 @@
+      },
+
+      '&[disabled]': {
+        border: theme.ButtonGroupButton_border_disabled,
+
+        '&[aria-checked=true]': {
+          backgroundColor:
+            theme.ButtonGroupButton_backgroundColor_checkedDisabled,
+          color: theme.ButtonGroupButton_color_checkedDisabled,
+
+          '&:hover': {
+            color: theme.ButtonGroupButton_color_checkedDisabled
+          }
+        }
+      }
+    },
+
+    '& > button:not(:first-child), & > *:not(:first-child) button': rtl
+      ? {
+          borderRightColor: 'transparent',
+          borderBottomRightRadius: 0,
+          borderTopRightRadius: 0
+        }
+      : {
+          borderLeftColor: 'transparent',
+          borderBottomLeftRadius: 0,
+          borderTopLeftRadius: 0
+        },
+
+    '& > button:not(:last-child), & > *:not(:last-child) button': rtl
+      ? {
+          borderBottomLeftRadius: 0,
+          borderTopLeftRadius: 0
+        }
+      : {
+          borderBottomRightRadius: 0,
+          borderTopRightRadius: 0
+        },
+
+    '& > [aria-checked]:not(:last-child)': rtl
+      ? { borderLeftColor: 'transparent' }
+      : { borderRightColor: 'transparent' },
+
+    '& > [aria-checked=false] + *:not([aria-checked]) button': rtl
+      ? { borderRightColor: theme.ButtonGroupButton_borderStartColor }
+      : { borderLeftColor: theme.ButtonGroupButton_borderStartColor },
+
+    '& > [aria-checked=true] + [aria-checked=true]': {
+      '&:not(:focus), & button:not(:focus)': {
+        borderLeftColor:
+          !rtl && theme.ButtonGroupButton_borderStartColor_checked,
+        borderRightColor:
+          rtl && theme.ButtonGroupButton_borderStartColor_checked
+      }
+    },
+
+    '& > [aria-checked=false], & > *:not([aria-checked])': {
+      '&:focus, & button:focus': {
+        borderLeftColor: theme.ButtonGroupButton_borderStartColor,
+        borderRightColor: theme.ButtonGroupButton_borderStartColor
+      }
+    },
+    '& > [aria-checked=false] + [aria-checked=false]': {
+      '&:not(:focus)': {
+        borderLeftColor: !rtl && theme.ButtonGroupButton_borderStartColor,
+        borderRightColor: rtl && theme.ButtonGroupButton_borderStartColor
+      }
+    }
+  };
+};
+
+const isChecked = (checked: number | Array<number> | Set<number>, index) => {
+  const isSet = checked instanceof Set;
+  const checkedSet = isSet ? checked : setFromArray(toArray(checked));
+  // $FlowFixMe - Refinement to Set not working
+  return checkedSet.has(index);
+};
+
+const Root = createStyledComponent('div', styles, {
+  displayName: 'ButtonGroup',
+  includeStyleReset: true
+});
+
+const getDefaultCheckedState = (props: Props) => {
+  const { children: _children, defaultChecked, mode } = props;
+  const children = Children.toArray(_children);
+
+  if (mode && defaultChecked !== undefined) {
+    const defaultCheckedArray = toArray(defaultChecked);
+    const checked =
+      mode === 'checkbox' ? defaultCheckedArray : [defaultCheckedArray[0]];
```

Comment:
I think `defaultCheckedArray.shift()` would be clearer than `[defaultCheckedArray[0]]`.

Reply:
https://devdocs.io/javascript/global_objects/array/shift

Reply:
I agree, this is not super clear, but shift returns the value that was removed & we need that to be in an array, so it would need to be `[defaultCheckedArray.shift()]`; is that more clear? Is it ok that shift permanently mutates the array here?

Reply:
Not super clear, but prolly not a big deal either, maybe just leave as is...